typing/env.ml

```diff
@@ -1267,11 +1265,16 @@ let mark_module_used name loc =
 let rec lookup_module_descr_aux ?loc ~mark lid env =
   match lid with
     Lident s ->
-      let (p, data) = IdTbl.find_name ~mark s env.components in
-      (p,
-       match data with
-       | Value (comp, _) -> comp
-       | Persistent -> (find_pers_struct s).ps_comps)
+      begin match IdTbl.find_name ~mark s env.components with
+      | exception Not_found ->
```

Comment:
I think you should add a guard checking that `s <> !current_unit` there. At least that's what the code was doing before #2041.